transit_odp/timetables/tasks.py

```diff
@@ -418,6 +443,50 @@ def task_dqs_upload(revision_id: int, task_id: int):
     N.B. this is just a proxy to `upload_dataset_to_dqs` as part of a refactor.
     """
     upload_dataset_to_dqs(task_id)
+
+
+@shared_task()
+def task_data_quality_service(revision_id: int, task_id: int) -> int:
+    """A task that runs the DQS checks on TxC file(s)."""
+    task = get_etl_task_or_pipeline_exception(task_id)
+    revision = task.revision
+    adapter = get_dataset_adapter_from_revision(logger=logger, revision=revision)
+    adapter.info("Starting DQS checks.")
+    try:
+        task.update_progress(95)
+        report = Report.initialise_dqs_task(revision)
+        checks = Checks.get_all_checks()
+        txc_file_attributes_objects = TXCFileAttributes.objects.for_revision(
+            revision_id
+        )
+        combinations = itertools.product(txc_file_attributes_objects, checks)
+
+        for txc_file_attribute, check in combinations:
+            TaskResults.initialize_task_results(report, txc_file_attribute, check)
```

Comment:
combinations can be passed to method `initialize_task_results` so that bulk_create can be used instead of bulk create for a set of checks for each file